eras/dijkstra/impl/testlib/Test/Cardano/Ledger/Dijkstra/Binary/Golden.hs

```diff
@@ -9,66 +9,117 @@
+    goldenDuplicateNativeScriptsDisallowed @era version
+    goldenDuplicatePlutusScriptsDisallowed @era version SPlutusV1
+    goldenDuplicatePlutusScriptsDisallowed @era version SPlutusV2
+    goldenDuplicatePlutusScriptsDisallowed @era version SPlutusV3
+    goldenDuplicatePlutusDataDisallowed @era version
+    goldenEmptyFields @era version
 
-duplicateCertsTx :: forall era. DijkstraEraTest era => Version -> Enc
-duplicateCertsTx v =
-  mconcat
-    [ E $ TkMapLen 4
-    , Em [E @Int 0, Ev v $ Set.empty @TxIn]
-    , Em [E @Int 1, Ev v $ [] @(TxOut era)]
-    , Em [E @Int 2, E $ Coin 0]
-    , Em
-        [ E @Int 4
-        , Em
-            [ E $ TkTag 258
-            , E $ TkListLen 2
-            , Ev v cert
-            , Ev v cert
-            ]
-        ]
-    ]
-  where
-    cert = DelegTxCert @era (KeyHashObj (mkKeyHash 0)) (DelegStake (mkKeyHash 1))
+goldenEmptyFields :: forall era. DijkstraEraTest era => Version -> Spec
+goldenEmptyFields version =
+  describe "Empty fields not allowed" $ do
+    describe "Untagged" $ do
+      let
+        expectFailureOnEmptyField k =
+          expectDecoderFailureAnn @(TxWits era) version (witsEmptyField k)
+      it "addrTxWits" . expectFailureOnEmptyField 0 $
+        DecoderErrorDeserialiseFailure
+          (Binary.label $ Proxy @(Annotator (TxWits era)))
+          (DeserialiseFailure 4 "Set cannot be empty")
+      it "nativeScripts" . expectFailureOnEmptyField 1 $
+        DecoderErrorCustom "Annotator" "Empty script Set is not allowed"
+      it "bootstrapWitness" . expectFailureOnEmptyField 2 $
+        DecoderErrorDeserialiseFailure
+          (Binary.label $ Proxy @(Annotator (TxWits era)))
+          (DeserialiseFailure 4 "Set cannot be empty")
+      it "plutusV1Script" . expectFailureOnEmptyField 3 $
+        DecoderErrorDeserialiseFailure
+          "Annotator (MemoBytes (AlonzoTxWitsRaw DijkstraEra))"
+          (DeserialiseFailure 4 "Empty list of scripts is not allowed")
+      it "plutusData" . expectFailureOnEmptyField 4 $
+        DecoderErrorCustom "Annotator" "Empty script Set is not allowed"
+      it "redeemers" . expectFailureOnEmptyField 5 $
+        DecoderErrorDeserialiseFailure
+          "Annotator (MemoBytes (AlonzoTxWitsRaw DijkstraEra))"
+          (DeserialiseFailure 2 "List encoding of redeemers not supported starting with PV 12")
+      it "plutusV2Script" . expectFailureOnEmptyField 6 $
+        DecoderErrorDeserialiseFailure
+          "Annotator (MemoBytes (AlonzoTxWitsRaw DijkstraEra))"
+          (DeserialiseFailure 4 "Empty list of scripts is not allowed")
+      it "plutusV3Script" . expectFailureOnEmptyField 7 $
+        DecoderErrorDeserialiseFailure
+          "Annotator (MemoBytes (AlonzoTxWitsRaw DijkstraEra))"
+          (DeserialiseFailure 4 "Empty list of scripts is not allowed")
+      -- TODO replace this with `plutusV4Script` once that is added
+      it "8th field" . expectFailureOnEmptyField 8 $
```

Comment:
Same suggestion about changing this to a property test. In the matter of fact, to avoid duplication it should be defined once and reused in all eras